### ISSUE TITLE
`ast.unparse` is needlessly "hip" for f-string quotes in Python 3.12+

### ISSUE THREAD

**Issue by nschloe** [python/cpython#127975]
# Feature or enhancement

### Proposal:

Up until Python 3.11, the output of the following `ast.parse`/`unparse` run

```python
import ast
from sys import version_info as vi

print(f"{vi.major}.{vi.minor}.{vi.micro}")
print(ast.unparse(ast.parse("f\"{'.' * 5}\"")))
```

was

```
3.11.11
f"{'.' * 5}"
```

In Python 3.12, [new f-string features were introduced](https://docs.python.org/3/whatsnew/3.12.html#pep-701-syntactic-formalization-of-f-strings), allowing for more general quote combinations. The output of the above script in Python 3.12 and 3.13 is

```
3.12.7
f'{'.' * 5}'
```

While this is legal Python 3.12/3.13, this representation needlessly restricts the usability of the generated code: It will not work on Python 3.11 and earlier.

I would thus like to suggest for `ast.unparse` to return, if possible, the more compatible code; in this case

```
f"{'.' * 5}"
```

which works across all currently supported Python versions.

### Has this already been discussed elsewhere?

No response given

### Linked PRs
* gh-127980
* gh-129600
* gh-129601

**Comment by picnixz** [comment]
One one's hand I would say "3.11 is oldstable so we shouldn't really care", but on the other hand, I don't think it should be that much work on our side to make it compatible. I can see some value for static analysis tools that would benefit from this but I don't know how much work is needed on our side.

cc @pablogsal @JelleZijlstra

**Comment by pablogsal** [comment]
I agree but the challenge here is that the unparse code is very sensitive to many of nodes and is very easy to fall into bugs such as https://github.com/python/cpython/issues/108843 or https://github.com/python/cpython/issues/108469 so it's a matter of checking if the fix is worth the maintenance cost.

@nschloe do you want to propose a PR so we can evaluate it?

**Comment by JelleZijlstra** [comment]
I think alternating quotes is better stylistically so it would be preferable to use the old style, but I agree with Pablo that we should do this only if it doesn't unduly complicate the implementation.

**Comment by hauntsaninja** [comment]
PR here: https://github.com/python/cpython/pull/127980

**Comment by nineteendo** [comment]
What will be the intended result of this? Should we use triple quotes?

```python
>>> print(ast.unparse(ast.parse("""f\"""{f'''{f"{''}"}'''}""\"""")))
f'''{f"""{f"{''}"}"""}'''
```

`repr(string)` doesn't use them (even though it could):

```python
>>> """f'''{f"{''}"}'''"""
'f\'\'\'{f"{\'\'}"}\'\'\''
```
